src/sbd_helper.in

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# Authors:
+#  Klaus Wenninger <kwenning@redhat.com>
+#
+# License: Revised BSD
+
+# chkconfig: - 99 01
+# description: Shared-storage based fencing daemon
+# processname: sbd
+#
+### BEGIN INIT INFO
+# Provides:		sbd_helper
+# Required-Start:	$local_fs $remote_fs
+# Should-Start:		$syslog
+# Required-Stop:	$local_fs $remote_fs
+# Default-Start:
+# Default-Stop:
+# Short-Description:	Starts and stops Shared-storage based fencing daemon.
+# Description:		Starts and stops Shared-storage based fencing daemon.
+### END INIT INFO
+
+desc="Shared-storage based fencing daemon"
+prog="sbd"
+pidfile="/var/run/${prog}.pid"
+
+# set secure PATH
+PATH="/sbin:/bin:/usr/sbin:/usr/bin:/usr/sbin"
+
+checkrc() {
+	if [ $? = 0 ]; then
+		success
+	else
+		failure
+	fi
+}
+
+success()
+{
+	echo -ne "[  OK  ]\r"
+}
+
+failure()
+{
+	echo -ne "[FAILED]\r"
+}
+
+log()
+{
+	logger -t sbd -p daemon.notice "$*"
+}
+
+notify()
+{
+	log "$*"
+	echo -n "   $*"
+}
+
+sbd_status()
+{
+	pid=$(cat $pidfile 2>/dev/null)
+	rtrn=$?
```

Comment:
This causes an issue b/c start calls status, and this will overwrite rtrn. You could set this local (which is what pacemaker does), but that's not actually portable across all POSIX shells (notably Solaris and derivatives). So a better approach would probably be either to use a unique name here (like status_rtrn), or explicitly set rtrn in start after calling status.